client/inprocess_test.go

```diff
@@ -22,13 +22,13 @@ func TestInProcessMCPClient(t *testing.T) {
 		"test-tool",
 		mcp.WithDescription("Test tool"),
 		mcp.WithString("parameter-1", mcp.Description("A string tool parameter")),
-		mcp.WithToolAnnotation(mcp.ToolAnnotation{
-			Title:           "Test Tool Annotation Title",
-			ReadOnlyHint:    true,
-			DestructiveHint: false,
-			IdempotentHint:  true,
-			OpenWorldHint:   false,
-		}),
+		mcp.WithToolAnnotation(
+			"Test Tool Annotation Title",
+			true,
+			false,
+			true,
+			false,
+		),
```

Comment:
This will be the UX for SDK users. It hides the pointer implementation from the users and it is more or less the same UX as we have now.

This follows similar `WithX` functions like:

```
WithResourceCapabilities(true, true),
WithPromptCapabilities(true),
WithToolCapabilities(true),
```